src/hooks/useLimitList.js

```diff
@@ -0,0 +1,12 @@
+import { useEffect, useState } from "react";
+import { getListOfProducts } from "../service/getProducts";
+
+export default function useLimitList(isLimited) {
+  const [list, setList] = useState([]);
+  useEffect(() => {
+    isLimited
+      ? getListOfProducts(setList, isLimited)
+      : getListOfProducts(setList);
```

Comment:
bad practice to set state outside component. call service method that returns data from Promise and update state with it